### DISCUSSION TITLE
How do whisper classify language?

### DISCUSSION THREAD

**Discussion by BaffledCoder** [openai/whisper#1615] [Q&A]
Does whisper directly classify language from audio, or does it first converts to text and then classify language?

**Comment by glangford** [comment]
If the language is not specified, whisper classifies the language using the first 30s of audio. The function to do that is here:

https://github.com/openai/whisper/blob/e8622f9afc4eba139bf796c210f5c01081000472/whisper/decoding.py#L19

**Reply by BaffledCoder** [reply]
Thanks for sharing, but I meant that whisper detect_langugae is based on classifying directly from the audio that is mel-features
Or does it requires text to predict language of audio

**Reply by glangford** [reply]
Directly from the first 30s of audio.
